internal/resource/config/authenticationpolicies/settings/authentication_policies_settings_resource.go

```diff
@@ -0,0 +1,227 @@
+type authenticationPoliciesSettingsResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+// GetSchema defines the schema for the resource.
+func (r *authenticationPoliciesSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Authentication Policies Settings",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the resource.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"enable_idp_authn_selection": schema.BoolAttribute{
+				Description: "Enable IdP authentication policies.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"enable_sp_authn_selection": schema.BoolAttribute{
+				Description: "Enable SP authentication policies.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalAuthenticationPoliciesSettingsFields(ctx context.Context, addRequest *client.AuthenticationPoliciesSettings, plan authenticationPoliciesSettingsModel) error {
+
+	if internaltypes.IsDefined(plan.EnableIdpAuthnSelection) {
+		addRequest.EnableIdpAuthnSelection = plan.EnableIdpAuthnSelection.ValueBoolPointer()
+	}
+
+	if internaltypes.IsDefined(plan.EnableSpAuthnSelection) {
+		addRequest.EnableSpAuthnSelection = plan.EnableSpAuthnSelection.ValueBoolPointer()
+	}
+
+	return nil
+
+}
+
+// Metadata returns the resource type name.
+func (r *authenticationPoliciesSettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_authentication_policies_settings"
+}
+
+func (r *authenticationPoliciesSettingsResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *authenticationPoliciesSettingsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationPoliciesSettingsModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createAuthenticationPoliciesSettings := client.NewAuthenticationPoliciesSettings()
+	err := addOptionalAuthenticationPoliciesSettingsFields(ctx, createAuthenticationPoliciesSettings, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for authentication policies settings", err.Error())
+		return
+	}
+	requestJson, err := createAuthenticationPoliciesSettings.MarshalJSON()
```

Comment:
We decided to remove all these marshalJsons from the code so this should be removed along with any other marshalljson calls. If we're going to continue to scaffold resources with the code generator we need to update it to leave this out.

Reply:
Still need to remove these